homeassistant/components/ads/sensor.py

```diff
@@ -25,21 +28,37 @@
 )
 
 DEFAULT_NAME = "ADS sensor"
+DEFAULT_DEVICE_CLASS = SensorDeviceClass.ENUM
+
 PLATFORM_SCHEMA = SENSOR_PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_ADS_VAR): cv.string,
         vol.Optional(CONF_ADS_FACTOR): cv.positive_int,
         vol.Optional(CONF_ADS_TYPE, default=ads.ADSTYPE_INT): vol.In(
             [
+                ads.ADSTYPE_BOOL,
+                ads.ADSTYPE_BYTE,
                 ads.ADSTYPE_INT,
                 ads.ADSTYPE_UINT,
-                ads.ADSTYPE_BYTE,
+                ads.ADSTYPE_SINT,
+                ads.ADSTYPE_USINT,
                 ads.ADSTYPE_DINT,
                 ads.ADSTYPE_UDINT,
+                ads.ADSTYPE_WORD,
+                ads.ADSTYPE_DWORD,
+                ads.ADSTYPE_LREAL,
+                ads.ADSTYPE_REAL,
+                ads.ADSTYPE_DATE,
+                ads.ADSTYPE_TIME,
+                ads.ADSTYPE_DATE_AND_TIME,
+                ads.ADSTYPE_TOD,
             ]
         ),
         vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
         vol.Optional(CONF_UNIT_OF_MEASUREMENT, default=""): cv.string,
+        vol.Optional(CONF_DEVICE_CLASS, default=DEFAULT_DEVICE_CLASS): vol.In(
+            [device_class.value for device_class in SensorDeviceClass]
+        ),
```

Comment:
Side note: I'm not sure that `unit_of_measurement` should have a default value... it's been known to cause issues in some integrations.
I'm also very surprised to see a default device class being set to ENUM. Again should it not default to None?
Finally, should state class be added?

https://github.com/home-assistant/core/blob/86ae70780ccbc58e0314e0f9e2ea26eb8a86d1a9/homeassistant/components/modbus/__init__.py#L362-L363

Reply:
removed unit of measurement for now, to simplify things for now. 
Enum was there because i was testing and it just left there as is.
Added State class and refactored the device class.

Reply:
The unit of measurement needs to be kept. It's just the default "" that needs to be removed I think.

> `vol.Optional(CONF_UNIT_OF_MEASUREMENT): cv.string,`

Reply:
added back in.